### PR TITLE
Fixed Issue #8 and #9

### PR DESCRIPTION
**Pull request by Fabio3rs**
Shared Variables aren't reset properly
0ABA causes heap corruption
Bug on some text opcodes when send pointer to it.

### CODE REVIEW

**Comment by thelink2012** [comment]
Will review it by the weekend.

File: CLEO4/CCustomOpcodeSystem.cpp [hunk 1]
```diff
@@ -1928,8 +1934,8 @@ namespace CLEO
 	//0ACE=-1,show_formatted_text_box %1d%
 	OpcodeResult __stdcall opcode_0ACE(CRunningScript *thread)
 	{
-		char fmt[128	];
-		char text[128];
+		char fmt[128] = { 0 };
```
Comment (thelink2012) [inline CLEO4/CCustomOpcodeSystem.cpp]: Hmmm, why are those things needed?

File: CLEO4/CCustomOpcodeSystem.cpp [hunk 2]
```diff
@@ -389,6 +389,13 @@ namespace CLEO
 		{
 			// process parameter as a pointer to string
 			GetScriptParams(thread, 1);
+
+			if (buf != nullptr)
```
Comment (thelink2012) [inline CLEO4/CCustomOpcodeSystem.cpp]: What does this do?

File: CLEO4/CCustomOpcodeSystem.cpp (hunk 1 above)
Comment (Fabio3rs) [reply CLEO4/CCustomOpcodeSystem.cpp]: Not necessarily. I removed them in the new commit.

File: CLEO4/CCustomOpcodeSystem.cpp (hunk 2 above)
Comment (Fabio3rs) [reply CLEO4/CCustomOpcodeSystem.cpp]: Copy the string sent (via pointer) to the opcode to buffer (the "fmt" variable in some CLEO's string related opcodes functions).

CLEO do it to string variables. (Look some lines bellow)

Comment (thelink2012) [reply CLEO4/CCustomOpcodeSystem.cpp]: Ah, I see what it does now.

**Comment by thelink2012** [comment]
Everything seems fine @x87, you may merge, publish and such.

**Comment by x87** [comment]
Thank you!

**Pull request merged**